desktop/desktopentry/desktopentry.go

```diff
@@ -0,0 +1,218 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package desktopentry
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+type DesktopEntry struct {
+	Filename string
+	Name     string
+	Icon     string
+	Exec     string
+
+	Hidden                bool
+	OnlyShowIn            []string
+	NotShownIn            []string
+	GnomeAutostartEnabled bool
+
+	Actions map[string]*Action
+}
+
+type Action struct {
+	Name string
+	Icon string
+	Exec string
+}
+
+type groupState int
+
+const (
+	unknownGroup groupState = iota
+	desktopEntryGroup
+	desktopActionGroup
+)
+
+func splitStringList(value string) []string {
+	return strings.FieldsFunc(value, func(r rune) bool { return r == ';' })
+}
+
+func Load(filename string) (*DesktopEntry, error) {
```

Comment:
this needs a doc comment, also we usually would call such a function just Read or ReadEntry